### DISCUSSION TITLE
New monthly release of PAC CLI 1.25 is public

### DISCUSSION THREAD

**Discussion by petrochuk** [microsoft/powerplatform-build-tools#404] [Announcements]
https://www.nuget.org/packages/Microsoft.PowerApps.CLI#readme-body-tab

solution
 - "smarter" solution import (GitHub issue #200 by HansMeiser999). Added new --skip-lower-version flag to solution import
 - export path is optional

 org
 - New list-settings command displays list of current environment settings
 - New update-settings command updates environment settings

 pcf
 - Fixed pcf push when pac is installed as a dotnet tool
 - Updated init command defaults
 - Solution customization prefix is now optional during push
 - pcf-scripts is fixed when generated types were either deleted or contain errors

 admin
 - Updated assign-user to not error on duplicate entry
 - Added support for creating app user and assigning role to pac admin create-service-principal
 - Added support for managed environments with set-governance-config
 - Deprecated commands are now hidden from help but still available

 modelbuilder
 - Removed --generateactions Command and Replaced with --generatesdkmessages
           Change is backward compatible if using a settings file to generate. shortcut for generatesdkmessages remains -a for compatibility
 - Added --emitentityetc control for emitting ETC code in code base,  Defaulted is now to not emit ETC on write.
 - Added --emitvirtualattributes control for emitting Virtual Attributes as part of ModelBuilding. This allows access to "name" properties that are stored in the formatedvalues property bag.
         added at the request of git: https://github.com/microsoft/powerplatform-vscode/issues/509
 - Updated Suppress Code Generation attribute behavior to remove tool version from files.
 - Updated generation of field enums to a partial class to support the request on Git: https://github.com/microsoft/powerplatform-vscode/discussions/558
 - Fixed issue where Warning about not writhing Proxy attribute was not emitted to the console.  https://github.com/microsoft/powerplatform-vscode/issues/516
 - Fixed issue where SDK message builder would miss messages due to the volume of data being read from Dataverse. https://github.com/microsoft/powerplatform-vscode/issues/413

 copilot
 - New list command displays list of AI Builder models
 - New predict command allows to test AI Builder models

 Misc
 - Fixed 'React PCF creation not working' (GitHub issue #584 by Ionavoss)
 - Increased Dataverse Service Client Timeout to default of 30 minutes.
 - Updated Microsoft.Identity.Client to 4.54.1
 - Updated Microsoft.PowerPlatform.Formulas.Tools to 0.6.3-preview
 - Updated Microsoft.Xrm.Tooling.Dmt libraries to 4.0.0.139

**Comment by sanjeev40084** [comment]
@petrochuk this is one jam packed release. kudos to the team. 👏
